### PR TITLE
`convert_to_numpy_parameters` uses `bind_new_parameters`

### LINKED ISSUE #2987: [BUG] pennylane.tape.Unwarp does not unwrap Hamiltonian coefficients
### Expected behavior

`qml.tape.unwrap` convert all ArrayBox in a tape to usual arrays.

### Actual behavior

Coefficients of the `Hamiltonian` object instance (accessible via `ham.coeffs`) are not unwrapped.

### Additional information

_No response_

### Source code

_No response_

### Tracebacks

_No response_

### System information

```shell
Name: PennyLane
Version: 0.26.0.dev0
Summary: PennyLane is a Python quantum machine learning library by Xanadu Inc.
Home-page: https://github.com/XanaduAI/pennylane
Author: 
Author-email: 
License: Apache License 2.0
Location: /home/chae-yeun/work/Xanadu/devel-py39/lib/python3.9/site-packages
Requires: appdirs, autograd, autoray, cachetools, networkx, numpy, pennylane-lightning, retworkx, scipy, semantic-version, toml
Required-by: PennyLane-Lightning

Platform info:           Linux-5.14.0-1042-oem-x86_64-with-glibc2.35
Python version:          3.9.13
Numpy version:           1.23.2
Scipy version:           1.9.0
Installed devices:
- lightning.qubit (PennyLane-Lightning-0.26.0.dev1)
- default.gaussian (PennyLane-0.26.0.dev0)
- default.mixed (PennyLane-0.26.0.dev0)
- default.qubit (PennyLane-0.26.0.dev0)
- default.qubit.autograd (PennyLane-0.26.0.dev0)
- default.qubit.jax (PennyLane-0.26.0.dev0)
- default.qubit.tf (PennyLane-0.26.0.dev0)
- default.qubit.torch (PennyLane-0.26.0.dev0)
- default.qutrit (PennyLane-0.26.0.dev0)
```

### Existing GitHub issues

- [X] I have searched existing GitHub issues to make sure the issue does not already exist.

### PR DESCRIPTION
**Pull request by albi3ro**
[sc-25029] Fixes #2987 

`convert_to_numpy_parameters` was created before `bind_new_parameters`, so it was still using the old way of `op.data = new_data`.

### CODE REVIEW

File: pennylane/transforms/convert_to_numpy_parameters.py [hunk 1]
```diff
@@ -15,34 +15,30 @@
...
 
+# pylint: disable=no-member
 def _convert_op_to_numpy_data(op: qml.operation.Operator) -> qml.operation.Operator:
     if math.get_interface(*op.data) == "numpy":
         return op
     # Use operator method to change parameters when it become available
-    copied_op = copy.copy(op)
-    copied_op.data = math.unwrap(op.data)
-    return copied_op
+    return qml.ops.functions.bind_new_parameters(op, math.unwrap(op.data))
```
Comment (mudit2812) [inline pennylane/transforms/convert_to_numpy_parameters.py:28]: Can we not use `tape.bind_new_parameters` now?

File: pennylane/transforms/convert_to_numpy_parameters.py [hunk 2]
```diff
@@ -15,34 +15,30 @@
...
-    # Use measurement method to change parameters when it becomes available
-    copied_m = copy.copy(m)
-    if isinstance(copied_m.obs, qml.operation.Tensor):
-        copied_m.obs.data = math.unwrap([o.data for o in m.obs.obs])
-    else:
-        copied_m.obs.data = math.unwrap(m.obs.data)
-    return copied_m
+        if m.eigvals is None or math.get_interface(m.eigvals) == "numpy":
+            return m
+        return type(m)(wires=m.wires, eigvals=math.unwrap(m.eigvals))
```
Comment (mudit2812) [inline pennylane/transforms/convert_to_numpy_parameters.py:38]: Great catch!

**Review by mudit2812 — APPROVED** [review]
Looks good! Just one question, but it's non-blocking

File: pennylane/transforms/convert_to_numpy_parameters.py (hunk 1 above)
Comment (albi3ro) [reply pennylane/transforms/convert_to_numpy_parameters.py:28]: Potentially yes, but then I would have to rewrite more things, and this gets the job done.

File: tests/transforms/test_convert_to_numpy_parameters.py
Comment (timmysilv) [inline tests/transforms/test_convert_to_numpy_parameters.py:1]: i suppose there are already some existing tests, but can you add specific coverage for the unique sub-cases now being handled separately?

File: pennylane/transforms/convert_to_numpy_parameters.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (albi3ro) [reply pennylane/transforms/convert_to_numpy_parameters.py:38]: Rewrote it slightly so we only check eigvals if `m.obs` is None. I forgot eigvals could also come from the observable.

**Review by timmysilv — APPROVED** [review]
once you add a test to please codecov for that one line, lgtm!

**Pull request merged**